src/helpers.rs

```diff
@@ -447,29 +447,77 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    fn read_os_str_from_target_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
+        if cfg!(target_os = "unix") {
+            self.read_os_str_from_c_str(scalar)
+        } else if cfg!(target_os = "windows") {
+            self.read_os_str_from_wide_str(scalar)
+        } else {
+            throw_unsup_format!("support for target OS not yet available")
+        }
+    }
+
     /// Helper function to read an OsString from a null-terminated sequence of bytes, which is what
     /// the Unix APIs usually handle.
-    fn read_os_str_from_c_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, &'a OsStr>
+    fn read_os_str_from_c_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString>
```

Comment:
I don't think this is strictly needed here, right? Just because `read_os_str_from_target_str` cannot work with an efficient reference, is no reason to make `read_os_str_from_c_str` less efficient.